allennlp/modules/seq2seq_decoders/stacked_self_attention_decoder_net.py

```diff
@@ -0,0 +1,149 @@
+from typing import Tuple, Dict, Optional
+import copy
+import math
+from overrides import overrides
+
+import torch
+from torch import nn
+from torch.autograd import Variable
+import torch.nn.functional as F
+
+from allennlp.modules.seq2seq_encoders.bidirectional_language_model_transformer import (subsequent_mask,
```

Comment:
Thanks for reusing this. Could you please credit the Annotated Transformer tutorial for the rest? http://nlp.seas.harvard.edu/2018/04/03/attention.html (If that's indeed where this is from.)